Unable to build Jupyter lab 3.0.14

Jupyter lab version= 3.0.14
python = 3.8.2
OS= Windows 10pro
After installing jupyter labextension spreadsheet jupyter lab failed to build. If i uninstall it then build successfully.

C:\Users\PC>jupyter labextension install jupyterlab-spreadsheet --no-build
-
C:\Users\PC>jupyter lab build --minimize=False
[LabBuildApp] JupyterLab 3.0.14
[LabBuildApp] Building in C:\Program Files\Autosphere\Process Studio\App\Python\share\jupyter\lab
[LabBuildApp] Building jupyterlab assets (production, not minimized)
Build failed.
Troubleshooting: If the build failed due to an out-of-memory error, you
may be able to fix it by disabling the `dev_build` and/or `minimize` options.

If you are building via the `jupyter lab build` command, you can disable
these options like so:

jupyter lab build --dev-build=False --minimize=False

You can also disable these options for all JupyterLab builds by adding these
lines to a Jupyter config file named `jupyter_config.py`:

c.LabBuildApp.minimize = False
c.LabBuildApp.dev_build = False

If you don't already have a `jupyter_config.py` file, you can create one by
adding a blank file of that name to any of the Jupyter config directories.
The config directories can be listed by running:

jupyter --paths

Explanation:

- `dev-build`: This option controls whether a `dev` or a more streamlined
`production` build is used. This option will default to `False` (i.e., the
`production` build) for most users. However, if you have any labextensions
installed from local files, this option will instead default to `True`.
Explicitly setting `dev-build` to `False` will ensure that the `production`
build is used in all circumstances.

- `minimize`: This option controls whether your JS bundle is minified
during the Webpack build, which helps to improve JupyterLab's overall
performance. However, the minifier plugin used by Webpack is very memory
intensive, so turning it off may help the build finish successfully in
low-memory environments.

An error occured.
RuntimeError: JupyterLab failed to build
See the log file for details:  C:\Users\PC\AppData\Local\Temp\jupyterlab-debug-9p9orm7d.log

Thanks for posting the debug log!

Checking the log, this appears to be a dependency issue with how JQuery is bundled:

```
31mModuleNotFoundError: Module not found: Error: Can't resolve '.\jquery.js' in 'C:\Program Files\Autosphere\Process Studio\App\Python\share\jupyter\lab\staging\node_modules\jquery\dist'
Did you mean './.\jquery.js'?
```

I had to hack in JQuery to get it to work within Jupyterlab's webpack build, and it seems likely that something in that hack broke. Investigating further, I'll see if I can repro this locally.

--- a/.gitignore
+++ b/.gitignore
@@ -3,3 +3,8 @@ node_modules/
 yarn-error.log
 # Jupyter
 .ipynb_checkpoints/
+
+# Misc
+.DS_Store
+# Excel's autosave format
+~$*

--- a/src/grid.ts
+++ b/src/grid.ts
@@ -4,7 +4,7 @@
 // carefully. It depends on JQuery and JQuery UI, so we load those first.
 // To get global symbols working, we need to use `expose-loader` to set the global symbol,
 // since we don't have control over the webpack config.
-import "expose-loader?jQuery!jquery";
+import "expose-loader?exposes=$,jQuery!jquery";
 import "jquery-ui";
 import "slickgrid/lib/jquery.event.drag-2.3.0.js";
 // Now we load SlickGrid's library folders, in dependency order

